ibm/resource_ibm_cis_page_rule.go

```diff
@@ -0,0 +1,418 @@
+	return &cisPageRuleValidator
+}
+
+func resourceCISPageRuleCreate(d *schema.ResourceData, meta interface{}) error {
+	cisClient, err := meta.(ClientSession).CisPageRuleClientSession()
+	if err != nil {
+		return err
+	}
+
+	crn := d.Get(cisID).(string)
+	zoneID, _, err := convertTftoCisTwoVar(d.Get(cisDomainID).(string))
+	cisClient.Crn = core.StringPtr(crn)
+	cisClient.ZoneID = core.StringPtr(zoneID)
+
+	targets := expandCISPageRuleTargets(d.Get(cisPageRuleTargets))
+	actions := expandCISPageRuleActions(d.Get(cisPageRuleActions))
+
+	opt := cisClient.NewCreatePageRuleOptions()
+	opt.SetTargets(targets)
+	opt.SetActions(actions)
+	if value, ok := d.GetOk(cisPageRulePriority); ok {
+		opt.SetPriority(int64(value.(int)))
+	}
+	if value, ok := d.GetOk(cisPageRuleStatus); ok {
+		opt.SetStatus(value.(string))
+	}
+
+	result, response, err := cisClient.CreatePageRule(opt)
+	if err != nil {
+		log.Printf("Create page rule failed: %v", response)
+		return err
+	}
+	d.SetId(convertCisToTfThreeVar(*result.Result.ID, zoneID, crn))
+	return resourceCISPageRuleRead(d, meta)
+}
+func resourceCISPageRuleRead(d *schema.ResourceData, meta interface{}) error {
+	cisClient, err := meta.(ClientSession).CisPageRuleClientSession()
+	if err != nil {
+		return err
+	}
+
+	ruleID, zoneID, crn, err := convertTfToCisThreeVar(d.Id())
+	if err != nil {
+		return err
+	}
+	cisClient.Crn = core.StringPtr(crn)
+	cisClient.ZoneID = core.StringPtr(zoneID)
+
+	opt := cisClient.NewGetPageRuleOptions(ruleID)
+	result, response, err := cisClient.GetPageRule(opt)
+	if err != nil {
+		log.Printf("Get page rule failed: %v", response)
+		return err
+	}
+	d.Set(cisID, crn)
+	d.Set(cisDomainID, zoneID)
+	d.Set(cisPageRuleID, *result.Result.ID)
+	d.Set(cisPageRulePriority, *result.Result.Priority)
+	d.Set(cisPageRuleStatus, *result.Result.Status)
+	d.Set(cisPageRuleTargets, flattenCISPageRuleTargets(result.Result.Targets))
+	d.Set(cisPageRuleActions, flattenCISPageRuleActions(result.Result.Actions))
+	return nil
+}
+func resourceCISPageRuleUpdate(d *schema.ResourceData, meta interface{}) error {
+	cisClient, err := meta.(ClientSession).CisPageRuleClientSession()
+	if err != nil {
+		return err
+	}
+
+	ruleID, zoneID, crn, _ := convertTfToCisThreeVar(d.Id())
+	cisClient.Crn = core.StringPtr(crn)
+	cisClient.ZoneID = core.StringPtr(zoneID)
+
+	if d.HasChange(cisPageRuleTargets) ||
+		d.HasChange(cisPageRuleActions) {
+
+		targets := expandCISPageRuleTargets(d.Get(cisPageRuleTargets))
+		actions := expandCISPageRuleActions(d.Get(cisPageRuleActions))
+
+		opt := cisClient.NewUpdatePageRuleOptions(ruleID)
+		opt.SetTargets(targets)
+		opt.SetActions(actions)
+		if value, ok := d.GetOk(cisPageRulePriority); ok {
+			opt.SetPriority(int64(value.(int)))
+		}
+		if value, ok := d.GetOk(cisPageRuleStatus); ok {
+			opt.SetStatus(value.(string))
+		}
+
+		_, response, err := cisClient.UpdatePageRule(opt)
```

Comment:
There are two features `update` and `change` in api docs are they both same? If not where is it handled in resource?

Reply:
both does the same.